homeassistant/components/govee_light_local/light.py

```diff
@@ -116,20 +119,14 @@ def rgb_color(self) -> tuple[int, int, int] | None:
     @property
     def color_mode(self) -> ColorMode | str | None:
         """Return the color mode."""
+        if self._fixed_color_mode:
+            return self._fixed_color_mode
         if (
             self._device.temperature_color is not None
             and self._device.temperature_color > 0
         ):
             return ColorMode.COLOR_TEMP
-        if self._device.rgb_color is not None and any(self._device.rgb_color):
-            return ColorMode.RGB
-
-        if (
-            self._attr_supported_color_modes
-            and ColorMode.BRIGHTNESS in self._attr_supported_color_modes
-        ):
-            return ColorMode.BRIGHTNESS
-        return ColorMode.ONOFF
+        return ColorMode.RGB
```

Comment:
I'm not sure if this is the correct way to detect color_temp mode or rgb mode.

Reply:
The API returns the value of the last set color mode (either RGB or Kelvin). However, there are devices that support only brightness, and I'm not sure about their response as I don't have one of them. 
I think it's probably better to keep the check on 'rgb_color' and use 'ColorMode.BRIGHTNESS' as a fallback.

Reply:
galorhallen: If a light does not support any adjustments its color mode should be set to `ColorMode.ONOFF`. If a light only supports brightness adjustments, its color mode should be set to `ColorMode.BRIGHTNESS`. With this PR, this now happens in the constructor.

The only case we need to dynamically set the color mode is if the light supports more than one color mode, the only way that can happen with this light is when the light supports both adjustable color temperature and RGB.

Reply:
Did you mean I should change from this code:
```py
        if (
            self._device.temperature_color is not None
            and self._device.temperature_color > 0
        ):
            return ColorMode.COLOR_TEMP
        return ColorMode.RGB
```
To this code:
        if self._device.rgb_color is not None and any(self._device.rgb_color):
            return ColorMode.RGB
        return ColorMode.COLOR_TEMP
```

The check `any(self._device.rgb_color)` looks a bit dodgy to be honest, it is false if the light is set to rgb color (0,0,0). What happens if you set the light to rgb color (0,0,0)?

Reply:
I ran a quick test against my device, and if the RGB is set to (0,0,0), the device reports RGB as (0,0,0), temperature as 0, and the 'on' flag is still set to 'on,' but the led strip is actually off.
I think I should handle this in the library in some way, maybe capping colors to 1 instead of 0.
Anyway, considering this, I think your proposed changes should be fine.

Reply:
Thanks a lot @Galorhallen 👍